presto-main/src/test/java/com/facebook/presto/type/TestBigintOperators.java

```diff
@@ -223,4 +225,41 @@ public void testCastFromVarchar()
         assertFunction("cast('37' as bigint)", BIGINT, 37L);
         assertFunction("cast('17' as bigint)", BIGINT, 17L);
     }
+
+    @Test
+    public void testOverflowAdd()
+            throws Exception
+    {
+        assertInvalidFunction(format("%s + 1", Long.MAX_VALUE), NUMERIC_VALUE_OUT_OF_RANGE);
+    }
+
+    @Test
+    public void testUnderflowSubtract()
+            throws Exception
+    {
+        long minValue = Long.MIN_VALUE + 1; // due to https://github.com/facebook/presto/issues/4571 MIN_VALUE solely cannot be used
+        assertInvalidFunction(format("%s - 2", minValue), NUMERIC_VALUE_OUT_OF_RANGE);
+    }
+
+    @Test
+    public void testOverflowMultiply()
+            throws Exception
+    {
+        assertInvalidFunction(format("%s * 2", Long.MAX_VALUE), NUMERIC_VALUE_OUT_OF_RANGE);
+        //assertInvalidFunction(format("%s * -1", Long.MIN_VALUE), NUMERIC_VALUE_OUT_OF_RANGE); TODO: uncomment when https://github.com/facebook/presto/issues/4571 is fixed
```

Comment:
I left this one commented out because I didn't want to extract it into a separate test method just to disable it but if you prefer this way I can do this.